airbyte-integrations/connectors/source-lever-hiring/source_lever_hiring/streams.py

```diff
@@ -22,74 +22,150 @@
-        """
-        yield {}
+        yield from response.json()["data"]
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        """Use Pydantic schema"""
+        return self.schema.schema()
 
 
 class IncrementalLeverHiringStream(LeverHiringStream, ABC):
+
+    state_checkpoint_interval = 100
+    cursor_field = "updatedAt"
+
+    def __init__(self, start_dt: int, **kwargs):
+        super().__init__(**kwargs)
+        self._start_ts = start_dt
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        state_ts = int(current_stream_state.get(self.cursor_field, 0))
+        return {self.cursor_field: max(latest_record.get(self.cursor_field), state_ts)}
+
+    def request_params(self, stream_state: Mapping[str, Any] = None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        state_ts = int(stream_state.get(self.cursor_field, 0))
+        params["updated_at_start"] = max(state_ts, self._start_ts)
+
+        return params
+
+
+class Opportunities(IncrementalLeverHiringStream):
     """
-    TODO fill in details of this class to implement functionality related to incremental syncs for your connector.
-         if you do not need to implement incremental sync for any streams, remove this class.
+    Opportunities stream: https://hire.lever.co/developer/documentation#list-all-opportunities
     """
 
-    # TODO: Fill in to checkpoint stream reads after N records. This prevents re-reading of data if the stream fails for any reason.
-    state_checkpoint_interval = None
+    schema = Opportunity
+    base_params = {"include": "followers", "confidentiality": "all"}
 
-    @property
-    def cursor_field(self) -> str:
-        """
-        TODO
-        Override to return the cursor field used by this stream e.g: an API entity might always use created_at as the cursor field. This is
-        usually id or date based. This field's presence tells the framework this in an incremental stream. Required for incremental.
 
-        :return str: The name of the cursor field.
-        """
-        return ""
+class Users(LeverHiringStream):
+    """
+    Users stream: https://hire.lever.co/developer/documentation#list-all-users
+    """
 
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
-        """
-        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest record and
-        the current state and picks the 'most' recent cursor. This is how a stream's state is determined. Required for incremental.
-        """
-        return {}
+    schema = User
+    base_params = {"includeDeactivated": True}
+
+
+class OpportynityChildStream(LeverHiringStream, ABC):
```

Comment:
can you create a follow up issue to add caching once it's enabled by default in the CDK?

Reply:
created #6352